### ISSUE TITLE
Beta 2.5.6 Crash

### ISSUE THREAD

**Issue by danjames92** [croneter/PlexKodiConnect#576]
See logs. Crash on playback. I did reset the database after upgrading from 2.5.5 -> 2.5.6.

[kodi_crashlog_20181222154827.log](https://github.com/croneter/PlexKodiConnect/files/2705345/kodi_crashlog_20181222154827.log)
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/2705346/kodi.log)

**Comment by croneter** [comment]
Try 2.5.7 :wink:

**Comment by croneter** [comment]
Ah, the usual darn Kodi bug. This crash is caused by Kodi caching artwork in the background. Let Kodi sit for a while, then you should be fine

**Comment by danjames92** [comment]
I saw it after I posted lol. 

I actually don't cache images at all. I have that option off. 2.5.7 is doing its initial sync now. Will report back.

**Comment by danjames92** [comment]
@croneter still happening after 2.5.7 upgrade (with db reset)

[kodi_crashlog_20181222172650.log](https://github.com/croneter/PlexKodiConnect/files/2705409/kodi_crashlog_20181222172650.log)
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/2705410/kodi.log)

**Comment by croneter** [comment]
Every video playback failes? 2.4.11 works?

**Comment by milnash** [comment]
Yes, both TV shows and movies are crashing when pressing play. Was working on version 2.5.5 and 2.5.6.

Don't know about 2.4.11 because it couldn't handle my large library.

Using Kodi 18 RC3

**Comment by danjames92** [comment]
@milnash I think I may have found the culprit and it might be a Kodi bug.

`Delay refresh rate change` set to any number of seconds seems to crash or freeze Kodi.

Set to off playback seems to be ok.

**Comment by danjames92** [comment]
I'm on LibreELEC 8.9.0.009 which is RC2 I believe.

**Comment by milnash** [comment]
Tried Kodi 18 RC2 now. No matter the settings, Kodi crash for me.
And I know Kodi 18 RC2 worked for me on PKC 2.5.5 and 2.5.6.

**Comment by craZy-C** [comment]
Cant see 2.5.7 in the add oms repository ?

**Comment by croneter** [comment]
@craZy-C share logs

**Comment by croneter** [comment]
To all, please be aware that LibreElec is still ALPHA for Kodi 18! I cannot possibly support Alpha or Nightly versions of Kodi as too much stuff can still break on the Kodi side

**Comment by craZy-C** [comment]
@croneter, did you get logs i shared last night or would you like some more

**Comment by craZy-C** [comment]
Noted, apologies. Have created NEW Issue

**Comment by danjames92** [comment]
@croneter The latest Librelec alpha is based on RC2. Are you saying this isn't something you will support?

**Comment by croneter** [comment]
I am saying that **LibreElec** is still in alpha state and thus potentially very instable, never mind the Kodi version it's using. Support will be rendered in due course once initial flaws have been ironed out and at the very least a beta version is out...

**Comment by milnash** [comment]
But PKC 2.5.5 works in Kodi 18 RC2 and RC3. I can try 2.5.7 tonight with Kodi 17.6 and get back to you.

**Comment by starman2908** [comment]
Hi @croneter Apologies, I posted in the wrong forum. I was on a break and just got home from night-shift, tis the season :) Anyways here are a couple of logs for you, one from a successful attempt at playback while PKC was still syncing ( https://1drv.ms/t/s!AjLrauIo92yxmMd7PCqVB-OGW6hOew ) - worth noting that kodi crashed on stopping playback. 

And this is the log of attempting playback after sync had finished and which crashed kodi. ( https://1drv.ms/t/s!AjLrauIo92yxmMd89HM6YgGe4BdoaA )

Hope it helps.

**Comment by starman2908** [comment]
This is quite odd...but playback is successful if i use alexa to initiate playback. Still crashed kodi on stopping though. Not sure if it helps but here is a log. https://1drv.ms/u/s!AjLrauIo92yxmMd-AlDFH_Q4gZV0IQ

**Comment by croneter** [comment]
Theoretically, a Python add-on should not be able to crash Kodi, especially if PKC is initiating playback like this:
```
DEBUG: PLEX.MOVIES: Full sys.argv received: ['plugin://plugin.video.plexkodiconnect.movies/', '2', '?plex_id=3571&plex_type=movie&mode=play&filename=Star.Trek.VI.The.Undiscovered.Country.1991.720p.BRrip.x264.YIFY.mp4', 'resume:false']
```
Could someone please take this to the Kodi team and the Kodi issue tracker? Thanks

**Comment by danjames92** [comment]
I'm not seeing anymore database locked messages. I am getting an intermittent problem when stopping playback where it just freezes up and I have to manually reboot. Honestly hard to say if it's a LibreELEC problem or PKC problem at this point. Will try and grab logs when it happens.

Do event logs cover this kind of thing or do I need a full debug log?

**Comment by croneter** [comment]
PKC hands over playback to Kodi. Please take this to the LibreElec guys!

**Comment by Norien** [comment]
I just updated to kodi RC 5.2 and PKC version 2.5.15. I did a full reset, imported my lib and attempted to play a video. kodi crashed at that point.

LOG: https://hastebin.com/rejaqicibu.sql

I've been seeing intermittently video's not playing for a couple versions now but it never crashed kodi.
